src/dotnet/APIView/APIViewWeb/Repositories/UsageSampleManager.cs

```diff
@@ -0,0 +1,119 @@
+            _sampleFilesRepository = sampleFilesRepository;
+            _commentsRepository = commentsRepository;
+        }
+
+        public async Task<List<UsageSampleModel>> GetReviewUsageSampleAsync(string reviewId)
+        {
+            return await _samplesRepository.GetUsageSampleAsync(reviewId);
+        }
+
+        public async Task<string> GetUsageSampleContentAsync(string fileId)
+        {
+            var file = await _sampleFilesRepository.GetUsageSampleAsync(fileId);
+
+            if(file == null) return null;
+
+            StreamReader reader = new StreamReader(file);
+            string htmlString = reader.ReadToEnd();
+
+            return htmlString;
+        }
+
+        public async Task<UsageSampleModel> UpsertReviewUsageSampleAsync(ClaimsPrincipal user, string reviewId, string sample, int revisionNum, string revisionTitle)
+        {
+            // markdig parser with syntax highlighting
+            MarkdownPipeline pipeline = new MarkdownPipelineBuilder()
+                .UseAdvancedExtensions()
+                .UseSyntaxHighlighting()
+                .Build();
+
+            string htmlSample = Markdown.ToHtml(sample, pipeline);
+
+            var stream = new MemoryStream(Encoding.UTF8.GetBytes(htmlSample));
+            var originalStream = new MemoryStream(Encoding.UTF8.GetBytes(sample));
+            UsageSampleModel SampleModel = new UsageSampleModel(user, reviewId, revisionNum);
+
+            SampleModel.RevisionTitle = revisionTitle;
+
+            // Create new file and upsert the updated model
+            UsageSampleFileModel SampleFile = new UsageSampleFileModel();
+            UsageSampleFileModel SampleOriginal = new UsageSampleFileModel();
+
+            SampleModel.Author = user.GetGitHubLogin();
+            SampleModel.UsageSampleFileId = SampleFile.UsageSampleFileId;
+            SampleModel.UsageSampleOriginalFileId = SampleOriginal.UsageSampleFileId;
+
+            await _samplesRepository.UpsertUsageSampleAsync(SampleModel);
+            await _sampleFilesRepository.UploadUsageSampleAsync(SampleModel.UsageSampleFileId, stream);
+            await _sampleFilesRepository.UploadUsageSampleAsync(SampleModel.UsageSampleOriginalFileId, originalStream);
+            return SampleModel;
+        }
+
+        public async Task<UsageSampleModel> UpsertReviewUsageSampleAsync(ClaimsPrincipal user, string reviewId, Stream fileStream, int revisionNum, string revisionTitle)
+        {
+            // For file upload. Read stream then continue.
+            StreamReader reader = new StreamReader(fileStream);
+            var sample = reader.ReadToEnd();
+            return await UpsertReviewUsageSampleAsync(user, reviewId, sample, revisionNum, revisionTitle);
+        }
+
+        public async Task DeleteUsageSampleAsync(ClaimsPrincipal user, string reviewId, string sampleId) 
+        {
+            var sampleModels = await _samplesRepository.GetUsageSampleAsync(reviewId);
```

Comment:
I would prefer to mark sample and comments as soft deleted using status flag instead of hard deleting it. This will help us reverting an accidental delete.